### ISSUE TITLE
Unable to add new folder with a name

### ISSUE THREAD

**Issue by XxBlackHatxX** [cryptomator/ios#74]
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/ios/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/ios/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

Unable to add a new folder and name it for the first time

### System Setup

- iOS: 14.7.1
- Cryptomator: 2.0.0 (502)
- …

### Steps to Reproduce

1. Open the files app
2. Open the vault and attempt to add a new folder.
3. Try to give the folder an a name

### Expected Behavior

When you hit down the folder should be created with the name provided.

### Actual Behavior

When you hit done you get an error message saying…  “cannot rename folder untitled folder. The file untitled folder couldn’t be renamed because it doesn’t exist”

### Reproducibility

Always

### Anything else?

This issue has happened specifically when you name the folder for the very first time. if you keep the default name and then rename it this doesn’t happen.

When I try to add a folder with a name while saving a file in the share sheet it does not cause any errors to happen. When I go back and check the files app the folder saved correctly.

I am using Cryptomator in the iCloud file provider if that helps.

**Comment by phil1995** [comment]
This is indeed a very strange bug. Because in my opinion, due to the path locking we use for online operations, this should only occur if the item has not yet been saved in the database. Because when a new folder is created via the Files app, first `createDirectory` is called with the default name `untitled folder` (after a threshold of a few seconds even if the user hasn't clicked done yet) and then a rename is performed if necessary. It should be noted that we have no influence on the process described in the last sentence. 

Unfortunately, I am also unable to reproduce this behavior. If you still have the log files, they could help us to investigate the behavior a bit more. You can export them as a zip file in the main app under Settings -> Send Log File.

**Comment by XxBlackHatxX** [comment]
So what I did yesterday is create a new vault and copied all the files to the new vault. So far both the old vault and the new one are working fine and creating folders without issue if this happens again I will sent the logs.

**Comment by tobihagemann** [comment]
Will close this issue for now but feel free to re-open it, thank you! If someone else can reproduce this issue, please let us know.

**Comment by XxBlackHatxX** [comment]
The issue started to happen again. Here are the logs at the time I created the folder.

What I did was. While at the root directory, I Create a new folder and attempted to name it “try 1”. It failed with the “folder does not exist” error. Then i open the folder which caused it to go back to root instead of opening an empty folder. What I do is which to another vault then back to this vault.  it will then allows me to change the name to try 2 and open the empty folder without it glitching back to the root directory .

This same behavior happens from this point forward creating any new folder. I need to leave the untitled folder default then back to a different directory and then coming back to the directory with the new folder before renaming it.

Something I noticed from the logs is when I try to name it the first time it does not show up in the logs. Mind you it’s not me renaming it just setting the name for the first time.

2021/09/13 08:53:01:987  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:01:995  FPExt: enumerator(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:01:995  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:01:996  enumerateItems called for identifier: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:01:997  readLock /(0x0000000139f28a80) called
2021/09/13 08:53:01:998  readLock /(0x0000000139f28a80) done
2021/09/13 08:53:01:998  readLock /(0x0000000139f302a0) called
2021/09/13 08:53:01:998  readLock /(0x0000000139f302a0) done
2021/09/13 08:53:02:067  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:02:069  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 935)) called
2021/09/13 08:53:02:069  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 935)) called
2021/09/13 08:53:02:071  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:02:071  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:05:288  FPExt: enumerator(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:05:288  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:05:292  enumerateItems called for identifier: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:05:292  FPExt: createDirectory(withName: untitled folder, inParentItemIdentifier: NSFileProviderRootContainerItemIdentifier) called
2021/09/13 08:53:05:298  readLock /(0x000000013b123180) called
2021/09/13 08:53:05:298  readLock /(0x000000013b123180) done
2021/09/13 08:53:05:298  readLock /(0x000000013b123fb0) called
2021/09/13 08:53:05:298  readLock /(0x000000013b123fb0) done
2021/09/13 08:53:05:299  readLock /(0x0000000139f10f90) called
2021/09/13 08:53:05:299  readLock /(0x0000000139f10f90) done
2021/09/13 08:53:05:300  readLock /(0x000000013b11f110) called
2021/09/13 08:53:05:300  readLock /(0x000000013b11f110) done
2021/09/13 08:53:05:300  writeLock /untitled folder(0x000000013b017490) called
2021/09/13 08:53:05:300  writeLock /untitled folder(0x000000013b017490) done
2021/09/13 08:53:05:300  writeLock /untitled folder(0x0000000139f15a50) called
2021/09/13 08:53:05:300  writeLock /untitled folder(0x0000000139f15a50) done
2021/09/13 08:53:05:311  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 936)) called
2021/09/13 08:53:05:311  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 936)) called
2021/09/13 08:53:05:312  FPExt: providePlaceholder(at: file:///private/var/mobile/Containers/Shared/AppGroup/74F0C718-8747-4857-B3FA-1FAA2710DBBC/File%20Provider%20Storage/FD723AC8-E591-4FF9-ABE0-713FB3D70233/936/untitled%20folder) called
2021/09/13 08:53:05:312  FPExt: persistentIdentifierForItem(at: file:///private/var/mobile/Containers/Shared/AppGroup/74F0C718-8747-4857-B3FA-1FAA2710DBBC/File%20Provider%20Storage/FD723AC8-E591-4FF9-ABE0-713FB3D70233/936/untitled%20folder) called
2021/09/13 08:53:05:312  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 936)) called
2021/09/13 08:53:05:369  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:05:542  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: 933)
2021/09/13 08:53:05:542  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:06:234  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:06:234  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:15:544  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 936)) called
2021/09/13 08:53:21:555  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 936)) called
2021/09/13 08:53:21:565  FPExt: enumerator(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:21:565  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:21:566  enumerateItems called for identifier: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:21:567  readLock /(0x0000000139f28a80) called
2021/09/13 08:53:21:567  readLock /(0x0000000139f28a80) done
2021/09/13 08:53:21:567  readLock /(0x0000000139f0abf0) called
2021/09/13 08:53:21:567  readLock /(0x0000000139f0abf0) done
2021/09/13 08:53:21:651  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:21:654  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:21:654  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:21:656  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:21:656  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:21:689  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:21:689  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:24:713  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:24:713  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:27:036  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:27:062  FPExt: enumerator(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:27:062  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)) called
2021/09/13 08:53:27:063  enumerateItems called for identifier: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:27:064  readLock /(0x0000000139f1d650) called
2021/09/13 08:53:27:064  readLock /(0x0000000139f1d650) done
2021/09/13 08:53:27:064  readLock /(0x0000000139d2b700) called
2021/09/13 08:53:27:064  readLock /(0x0000000139d2b700) done
2021/09/13 08:53:27:138  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:27:143  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:27:143  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 917)) called
2021/09/13 08:53:40:920  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:40:922  FPExt: renameItem(withIdentifier: 937, toName: Try 2) called
2021/09/13 08:53:40:930  readLock /(0x0000000139d0b310) called
2021/09/13 08:53:40:930  readLock /(0x0000000139d0b310) done
2021/09/13 08:53:40:930  readLock /(0x000000013b11f3b0) called
2021/09/13 08:53:40:930  readLock /(0x000000013b11f3b0) done
2021/09/13 08:53:40:931  readLock /(0x000000013b121940) called
2021/09/13 08:53:40:931  readLock /(0x000000013b121940) done
2021/09/13 08:53:40:931  readLock /(0x0000000139d21ba0) called
2021/09/13 08:53:40:931  readLock /(0x0000000139d21ba0) done
2021/09/13 08:53:40:932  writeLock /untitled folder(0x0000000139d29be0) called
2021/09/13 08:53:40:932  writeLock /untitled folder(0x0000000139d29be0) done
2021/09/13 08:53:40:932  writeLock /untitled folder(0x0000000139d2a300) called
2021/09/13 08:53:40:932  writeLock /untitled folder(0x0000000139d2a300) done
2021/09/13 08:53:40:932  writeLock /Try 2(0x000000013b012eb0) called
2021/09/13 08:53:40:932  writeLock /Try 2(0x000000013b012eb0) done
2021/09/13 08:53:40:933  writeLock /Try 2(0x000000013b0115f0) called
2021/09/13 08:53:40:933  writeLock /Try 2(0x000000013b0115f0) done
2021/09/13 08:53:40:955  FPExt: urlForItem(withPersistentIdentifier: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:40:955  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:40:956  FPExt: providePlaceholder(at: file:///private/var/mobile/Containers/Shared/AppGroup/74F0C718-8747-4857-B3FA-1FAA2710DBBC/File%20Provider%20Storage/FD723AC8-E591-4FF9-ABE0-713FB3D70233/937/Try%202) called
2021/09/13 08:53:40:956  FPExt: persistentIdentifierForItem(at: file:///private/var/mobile/Containers/Shared/AppGroup/74F0C718-8747-4857-B3FA-1FAA2710DBBC/File%20Provider%20Storage/FD723AC8-E591-4FF9-ABE0-713FB3D70233/937/Try%202) called
2021/09/13 08:53:40:956  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:41:162  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: 933)
2021/09/13 08:53:41:162  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: NSFileProviderRootContainerItemIdentifier)
2021/09/13 08:53:45:728  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:45:735  FPExt: enumerator(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:45:735  FPExt: item(for: NSFileProviderItemIdentifier(_rawValue: 937)) called
2021/09/13 08:53:45:738  enumerateItems called for identifier: NSFileProviderItemIdentifier(_rawValue: 937)
2021/09/13 08:53:45:739  readLock /(0x0000000139d29be0) called
2021/09/13 08:53:45:739  readLock /(0x0000000139d29be0) done
2021/09/13 08:53:45:739  readLock /Try 2(0x000000013b2197a0) called
2021/09/13 08:53:45:739  readLock /Try 2(0x000000013b2197a0) done
2021/09/13 08:53:45:739  readLock /Try 2(0x000000013b11e5a0) called
2021/09/13 08:53:45:739  readLock /Try 2(0x000000013b11e5a0) done
2021/09/13 08:53:45:757  FPExt: enumerate now changes for: NSFileProviderItemIdentifier(_rawValue: 937)

**Comment by tobihagemann** [comment]
Thank you for giving more examples and the logs. I'm still trying to reproduce this issue without any success. Let's keep this issue open until we can figure out what's going on.

**Comment by XxBlackHatxX** [comment]
If it helps I'm using iCloud as my provider. I had worked on the same vault yesterday on my Mac running the newest version of Cryptomator. I don't know if there is something about working on the desktop version of Cryptomator causes this issue to start. I had worked on this vault using only with my iPhone and iPad until yesterday. That was the first time using it on my Mac after creating this new vault. I didn't even create any new folders on my Mac only opened a few PDFs and added some word docs to one of the folder.
